lib/resolveTaskFn.js

```diff
@@ -101,9 +104,29 @@ export const resolveTaskFn = ({
   debugLog('execaOptions:', execaOptions)
 
   return async (ctx = getInitialState()) => {
-    const result = await (shell
+    let isExecutionDone = false
+    const execaChildProcess = shell
       ? execa.command(isFn ? command : `${command} ${files.join(' ')}`, execaOptions)
-      : execa(cmd, isFn ? args : args.concat(files), execaOptions))
+      : execa(cmd, isFn ? args : args.concat(files), execaOptions)
+
+    const interruptExecutionOnError = async () => {
+      if (!isExecutionDone) {
+        if (ctx.errors.size > 0) {
+          const ids = await pidTree(execaChildProcess.pid)
+          ids.forEach(process.kill)
+
+          // The execa process is killed separately in order
+          // to get the `KILLED` status.
+          execaChildProcess.kill()
+        } else {
+          setTimeout(interruptExecutionOnError, FAIL_CHECK_INTERVAL)
```

Comment:
Maybe a bit confusing to use `setTimeout` with a variable called "interval", is it intended to be `setInterval`?

Reply:
Not with my original implementation. But this made me realize that it's better to restructure in a way that `setInterval` can be used. 👍